### PR TITLE
ddl: fix canceling model.ActionRebaseAutoID and model.ActionShardRowID ddl jobs

### PR DESCRIPTION
**Pull request by winkyao**
### What problem does this PR solve? 

Ref #8171, Fix the problem that cancel the model.ActionRebaseAutoID and model.ActionShardRowID jobs.

### What is changed and how it works?

Only allow to cancel them when the job is not handled. Otherwise, the cancel operation will be rejected.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by winkyao** [comment]
@crazycs520 @zimulala @ciscoxll PTAL

File: ddl/rollingback.go [hunk 1]
```diff
@@ -205,6 +205,25 @@ func rollingbackRenameIndex(t *meta.Meta, job *model.Job) (ver int64, err error)
 	return ver, errors.Trace(err)
 }
 
+func cancelOnlyNotHandledJob(job *model.Job) (ver int64, err error) {
+	// We can only cancel the not handled rebase auto id ddl job.
```
Comment (zimulala) [inline ddl/rollingback.go]: Do we need to update this comment?

File: ddl/ddl_worker_test.go [hunk 2]
```diff
@@ -481,6 +487,8 @@ func (s *testDDLSuite) TestCancelJob(c *C) {
 			return
 		}
 	}
+	tc.onJobUpdated = hookCancelFunc
+	tc.onJobRunBefore = hookCancelFunc
```
Comment (zimulala) [inline ddl/ddl_worker_test.go]: Why change this code?

File: ddl/ddl_worker_test.go [hunk 3]
```diff
@@ -325,7 +325,7 @@ func checkCancelState(txn kv.Transaction, job *model.Job, test *testCancelJob) e
 	// If the action is adding index and the state is writing reorganization, it wants to test the case of cancelling the job when backfilling indexes.
 	// When the job satisfies this case of addIndexFirstReorg, the worker hasn't started to backfill indexes.
```
Comment (zimulala) [inline ddl/ddl_worker_test.go]: Please update these comments.

File: ddl/ddl_worker_test.go (hunk 2 above)
Comment (winkyao) [reply ddl/ddl_worker_test.go]: we need to hook before the job running.

Comment (zimulala) [reply ddl/ddl_worker_test.go]: I know the function of `onJobRunBefore`, but why we need to add this hook to it?

File: ddl/ddl_worker_test.go (hunk 3 above)
Comment (winkyao) [reply ddl/ddl_worker_test.go]: Done

**Comment by winkyao** [comment]
@zimulala PTAL

File: ddl/ddl_worker_test.go (hunk 2 above)
Comment (winkyao) [reply ddl/ddl_worker_test.go]: If not, you can not hook the job when the state is None. when the job updated, the job.State is definitely not None.

**Comment by zimulala** [comment]
Please cherry-pick to v2.1

**Pull request merged**